app/models/embedded_ansible_worker/runner.rb

```diff
@@ -60,13 +60,21 @@ def do_exit(*args)
     Thread.exit
   end
 
-  def update_embedded_ansible_manager
-    ansible = ManageIQ::Providers::EmbeddedAnsible::AutomationManager.first_or_initialize
-    server  = MiqServer.my_server(true)
-    ansible.default_endpoint.url = URI::HTTPS.build(:host => server.hostname, :path => "/ansibleapi/v1")
-    ansible.name = "Embedded Ansible"
-    ansible.zone = server.zone
-    ansible.save!
+  def update_embedded_ansible_provider
+    server   = MiqServer.my_server(true)
+    provider = ManageIQ::Providers::EmbeddedAnsible::Provider.first_or_initialize
+
+    provider.name = "Embedded Ansible"
+    provider.zone = server.zone
+    provider.url  = URI::HTTPS.build(:host => server.hostname || server.ipaddress, :path => "/ansibleapi/v1").to_s
```

Comment:
Yeah, I'm thinking we can have something like [this](https://github.com/ManageIQ/manageiq/blob/7194f99b79815413e6cf75d42b6e078f678ff55f/app/models/manageiq/providers/ansible_tower/provider.rb#L66) for EmbeddedAnsible::Provider: